src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -256,6 +257,24 @@ private function getNormalizer(mixed $data, ?string $format, array $context): ?N
                     continue;
                 }
 
+                if ($normalizer instanceof SupportedTypesMethodInterface && (null !== $supportedTypes = $normalizer->getSupportedTypes())) {
+                    foreach ($supportedTypes as $supportedType => $isCacheable) {
+                        if ($type !== $supportedType && !is_subclass_of($type, $supportedType, true)) {
```

Comment:
Why not

```suggestion
                        if (!is_a($type, $supportedType, true)) {
```

Reply:
the current check is faster, and more importantly, it's compatible with `native-*` special types